src/swarm-tui/Swarm/TUI/View.hs

```diff
@@ -1540,11 +1540,18 @@ renderREPLPrompt focus theRepl = ps1 <+> replE
  where
   prompt = theRepl ^. replPromptType
   replEditor = theRepl ^. replPromptEditor
-  color = if theRepl ^. replValid then id else withAttr redAttr
+  color t =
+    if theRepl ^. replValid
+      then txt t
+      else case theRepl ^. replErrorSrcLoc of
+        NoLoc -> withAttr redAttr (txt t)
+        SrcLoc s e ->
+          let (validL, (invalid, validR)) = T.splitAt (e - s) <$> T.splitAt s t
```

Comment:
A common situation that I think we need a special case for is when there is a syntax error right at the very end.  This frequently happens when you have mismatched parens or a missing close quote etc.  Currently, in that case nothing gets highlighted at all, since the `invalid` part after the start location of the error (as well as `validR`) is the empty string.

Perhaps we can check whether the `invalid` part is the empty string (or only whitespace) and in that case highlight the entire REPL input?

Reply:
And yes, I was going to comment that this should be `e - s` instead of just `e` but you figured it out before me. :smile:

Reply:
Good idea, done.